php/proxy.php

```diff
@@ -3,150 +3,192 @@
  * Acrolinx Proxy
  * Description - Creates proxy for Acrolinx.
  * Author - Acrolinx
- * Version - 1.0
+ * Version - 1.1
  */
- 
+
 class Proxy {
+
     private $settings;
     private $targetPath;
     private $username;
     private $password;
-    private $domainURL;	
-	
+    private $domainURL;
+    private $host;
+    private $url;
+
     private function disable_gzip() {
         @ini_set('zlib.output_compression', 'Off');
-        @ini_set('output_handler', '');	
+        @ini_set('output_handler', '');
     }
-    
-    
+
     private function getSettings() {
         $acrolinxSettings = array();
-
         $acrolinxSettings['isSSO'] = 1; //TODO: Set if SSO should be used, otherwise it will be just a reverse proxy.
         $acrolinxSettings['url'] = ""; //TODO: Set Acrolinx server url here.
         $acrolinxSettings['genericPassword'] = "secret"; //TODO: Set secret token here.
 
         if($acrolinxSettings['isSSO'] == 1) {
-
             $acrolinxSettings['password'] = $acrolinxSettings['genericPassword'];
             $acrolinxSettings['username'] = get_current_user(); //TODO: Implement get_current_user, which retrieves the username from the current session.
         }
         return $acrolinxSettings;
     }
-    
+
     private function getPosition($requestURI, $part) {
         return strpos($requestURI, $part);
     }
-    
+
     private function getActualURI($requestURI, $pos, $part) {
         return substr_replace($requestURI, '', 0, ($pos+strlen($part)));
     }
-    
+
+    private function emulate_getallheaders() {
+        foreach ($_SERVER as $name => $value) {
+            if (substr($name, 0, 5) == 'HTTP_') {
+                $name = str_replace(' ', '-', ucwords(strtolower(str_replace('_', ' ', substr($name, 5)))));
+                $headers[$name] = $value;
+            } else if ($name == "CONTENT_TYPE") {
+                $headers["Content-Type"] = $value;
+            } else if ($name == "CONTENT_LENGTH") {
+                $headers["Content-Length"] = $value;
+            }
+        }
+        return $headers;
+    }
+
     private function getHeadersOfRequest() {
         $headers = array();
-        $isAuthToken = false; 
-        foreach (getallheaders() as $name => $value) {
-          $headerString = $name.':'.$value;
-
-          //Setting target Server as Host
-          if(strtolower($name) == 'host') {
-              $headerString = $name.':'.parse_url(self::getSettings()['url'])['host'];
-          }
-
-          if($name == 'authToken') {
-            $isAuthToken = true;
-          }
-          array_push($headers,"$headerString");
+        $isAuthToken = false;
+
+        foreach (self::emulate_getallheaders() as $name => $value) {
+
+            $headerString = $name.':'.$value;
+
+            //Setting target Server as Host
+            if(strtolower($name) == 'host') {
```

Comment:
It looks like we have a couple of these checks for headers. Shall we extract a method to do so?

Reply:
Yes there are two loops on headers.
Fix:  I have made the emulated getallheaders function as a fallback to native getallheaders